pandas/core/common.py

```diff
@@ -225,6 +225,10 @@ def asarray_tuplesafe(values, dtype=None):
     if isinstance(values, list) and dtype in [np.object_, object]:
         return construct_1d_object_array_from_listlike(values)
 
+    if isinstance(values, list) and hasattr(values, "__array__"):
+        # avoid converting extension array to numpy array
```

Comment:
so we're looking at some custom list subclass?

Reply:
@jbrockmendel 
"asarray_tuplesafe" and "_convert_arr_indexer" both return "array-like" object. before, "asarray_tuplesafe" and "_convert_arr_indexer" return numpy array type. But i think, both can return  EA type also

Reply:
why can't you use use is_list_like?